pkg/lightning/backend/local/duplicate.go

```diff
@@ -566,7 +601,21 @@ func (m *dupeDetector) RecordIndexConflictError(ctx context.Context, stream DupK
 		if err != nil {
 			return errors.Trace(err)
 		}
-		m.hasDupe.Store(true)
+
+		if currentKey == nil || !bytes.Equal(currentKey, key) {
+			if err := flushCurrentGroup(); err != nil {
```

Comment:
Please add a comment to explain why we need to do it

Reply:
53ef05f